File: vibe-coding/todo-app/frontend_web/src/modules/tasks_mod/application/search_tasks/SearchTasksResultDto.ts
import type { TaskEntity } from "@/modules/tasks_mod/domain/entities/TaskEntity";
import type { SearchTasksResponse } from "@/modules/tasks_mod/infrastructure/repositories/TasksReaderApiRepository";

/** Salida del caso de uso SearchTasks. Traduce snake_case -> camelCase. */
export class SearchTasksResultDto {
    private constructor(
        public readonly items: TaskEntity[],
        public readonly total: number,
    ) {}

    public static fromPrimitives(response: SearchTasksResponse): SearchTasksResultDto {
        const items: TaskEntity[] = response.items.map((row) => ({
            id: row.id,
            idList: row.id_list,
            title: row.title,
            description: row.description,
            isDone: row.is_done,
            dueDate: row.due_date,
            position: row.position,
        }));
        return new SearchTasksResultDto(items, response.total);
    }
}

File: vibe-coding/todo-app/backend_web/src/modules/tasks_mod/application/search_tasks/search_tasks_result_dto.py
from dataclasses import dataclass
from typing import Any, Self, final

from src.modules.tasks_mod.domain.enums.task_field_enum import TaskFieldEnum


@final
@dataclass(frozen=True, slots=True)
class SearchTasksResultDto:
    """Salida del caso de uso SearchTasks."""

    items: list[dict[str, Any]]
    total: int

    @classmethod
    def from_primitives(cls, primitives: dict[str, Any]) -> Self:
        items = list(primitives.get(TaskFieldEnum.ITEMS, []))
        return cls(items=items, total=len(items))

    def to_dict(self) -> dict[str, Any]:
        return {
            TaskFieldEnum.ITEMS: self.items,
            TaskFieldEnum.TOTAL: self.total,
        }
